### PR TITLE
add bucket_exists() to s3 wrapper

### LINKED ISSUE #275: ZMON S3 function to check for bucket existence
Usage of [.head_object(Bucket='string')](http://boto3.readthedocs.io/en/latest/reference/services/s3.html#S3.Client.head_bucket) makes this easy. Right now, the workaround is using the provided ZMON function `.list_bucket(bucket_name, '', 1)` and check for `client.exceptions.NoSuchBucket` or similar, which is not nice.

### PR DESCRIPTION
**Pull request by vetinari**
fixes #275

### CODE REVIEW

**Comment by vetinari** [comment]
:+1:

File: tests/test_s3.py [hunk 1]
```diff
@@ -176,3 +176,27 @@ def writer_side_effect(*args, **kwargs):
...
+def test_bucket_exists(monkeypatch):
+    client = MagicMock()
+    client.head_bucket.return_value = {}
+    get = MagicMock()
+    get.return_value.json.return_value = {'region': 'eu-central-1'}
+    monkeypatch.setattr('requests.get', get)
+    monkeypatch.setattr('boto3.client', lambda x, region_name: client)
+    s3_wrapper = S3Wrapper()
+
+    assert s3_wrapper.bucket_exists('foo') is True
```
Comment (mohabusama) [inline tests/test_s3.py]: could be useful to add ``client.head_bucket.assert_called_with()`` here

**Comment by vetinari** [comment]
:+1:

**Comment by mohabusama** [comment]
👍

**Pull request merged**